lib/discourse.rb

```diff
@@ -215,31 +215,52 @@ def self.base_url
     base_url_no_prefix + base_uri
   end
 
-  def self.enable_readonly_mode
-    $redis.set(readonly_mode_key, 1)
+  MAINTENANCE_READONLY_MODE_KEY ||= "readonly_mode:maintenance".freeze
+  READONLY_MODE_KEY_TTL ||= 60
```

Comment:
I feel like this TTL is too short to survive a rebuild but I guess we've not gotten bug reports about this yet.